sdk/storage/Azure.Storage.Blobs/AzureStorageNetMigrationV12.md

```diff
@@ -317,58 +323,221 @@ BlobClient blobClient = containerClient.GetBlobClient(blobName);
+foreach (IListBlobItem item in results)
+{
+    // process blob listing
+}
+```
+Note there is no asynchronous overload of a lazy enumerator in v11. Users desiring async performance were required to use manual page iteration and asynchronously request each page.
+
+v11 manual interation
 ```csharp
-// List the blobs in the container.
-// Assumes a reference to the container via `cloudBlobContainer`
+CloudBlobContainer cloudBlobContainer; // properly configured reference to an existing container
+// set this to already existing continuation token to pick up where you previously left off
 BlobContinuationToken blobContinuationToken = null;
 do
 {
-    var results = await cloudBlobContainer.ListBlobsSegmentedAsync(null, blobContinuationToken);
-    // Get the value of the continuation token returned by the listing call.
-    blobContinuationToken = results.ContinuationToken;
-    foreach (IListBlobItem item in results.Results)
+    BlobResultSegment resultSegment = await cloudBlobContainer.ListBlobsSegmentedAsync(blobContinuationToken);
+    blobContinuationToken = resultSegment.ContinuationToken;
+    foreach (IListBlobItem item in resultSegment.Results)
     {
-        Console.WriteLine(item.Uri);
+        // process blob listing
     }
 } while (blobContinuationToken != null); // Loop while the continuation token is not null.
 ```
 
+v12 lazy enumerable
+```csharp
+BlobContainerClient containerClient; // properly configured client to an existing container
+IAsyncEnumerable<BlobItem> results = containerClient.GetBlobsAsync());
+await foreach (BlobItem item in results)
+{
+    // process blob listing
+}
+```
+
+v12 manual iteration
+
+The result we declared an `IAsyncEnumerable<T>` in the previous example was actually an `AsyncPagable<T>`, an implementation provided by the Azure.Core package. This class contains the method `AsPages()`, which returns an `IAsyncEnumerable<Page<T>>`. This is how you can go page by page in the modern Storage SDK.
+
+```csharp
+BlobContainerClient containerClient; // properly configured client to an existing container
+// set this to already existing continuation token to pick up where you previously left off
+string initialContinuationToken = null;
+AsyncPageable<BlobItem> results = containerClient.GetBlobsAsync());
+IAsyncEnumerable<Page<BlobItem>> pages =  results.AsPages(initialContinuationToken);
+
+// the foreach loop requests the next page of results every loop
+// you do not need to explicitly access the continuation token just to get the next page
+// to stop requesting new pages, break from the loop
+// you also have access to the contination token returned with each page if needed
+await foreach (Page<BlobItem> page in pages)
+{
+    // process page
+
+    foreach (BlobItem item in page.Values)
+    {
+        // process blob listing
+    }
+
+    // access continuation token if desired
+    string continuationToken = page.ContinuationToken;
```

Comment:
might be worth adding sample how to start iteration from continuation token. I had few customers asking for it as they couldn't discover that.

Reply:
```csharp
// set this to already existing continuation token to pick up where you previously left off
string initialContinuationToken = null;
```
That's the intent of these lines at the start of this sample. Perhaps we need to better highlight this section? Or rephrase?

Reply:
no it's good. I'm blind. thanks!